core/src/main/java/org/infinispan/interceptors/compat/BaseTypeConverterInterceptor.java

```diff
@@ -128,19 +138,24 @@ protected void init(InternalEntryFactory entryFactory, VersionGenerator versionG
       if (ctx.isOriginLocal()) {
          command.setKey(converter.boxKey(key));
       }
-      Object ret = ctx.forkInvocationSync(command);
-      if (ret != null) {
-         CacheEntry entry = (CacheEntry) ret;
+      ctx.onReturn((ctx1, command1, rv, throwable) -> {
+         if (throwable != null)
+            throw throwable;
+
+         if (rv == null) {
+            return null;
+         }
+         CacheEntry entry = (CacheEntry) rv;
          Object returnValue = entry.getValue();
          if (needsUnboxing(ctx)) {
             returnValue = converter.unboxValue(entry.getValue());
          }
          // Create a copy of the entry to avoid modifying the internal entry
-         return ctx.shortCircuit(entryFactory.create(
-               entry.getKey(), returnValue, entry.getMetadata(),
-               entry.getLifespan(), entry.getMaxIdle()));
-      }
-      return ctx.shortCircuit(null);
+         return CompletableFuture.completedFuture(entryFactory
```

Comment:
I think i can skip it and merge the if statements, same as for `GetKeyValueCommand`.